spec/requests/api/v1/search_spec.rb

```diff
@@ -1,106 +0,0 @@
-    it 'returns a 401 with an error message' do
-      headers = { 'Authorization' => 'invalid' }
-      get '/v1/search', headers: headers, as: :json
-      expect(response.code).to eq '401'
-      expect(response.body).to eq 'auth is wrong'
-    end
-  end
-
-  context 'with a good token' do
-    let(:jwt_token) do
-      JWT.encode(
-        { aud: Rails.application.secrets.artsy_application_id },
-        Rails.application.secrets.artsy_internal_secret
-      )
-    end
-
-    let(:headers) { { 'Authorization' => "Bearer #{jwt_token}" } }
-
-    context 'with no term' do
-      it 'returns a 400 with an error message' do
-        get '/v1/search', headers: headers, as: :json
-
-        expect(response.code).to eq '400'
-        expect(response.body).to eq 'no term parameter'
-      end
-    end
-
-    context 'with a valid term' do
-      it 'returns the matches for that term' do
-        Organization.recreate_index!
-        organization = Fabricate :organization
-        Fabricate :organization_name, organization: organization, content: 'David Zwirner Gallery'
-        organization.es_index
-        Organization.refresh_index!
-
-        get '/v1/search', params: { term: 'David' }, headers: headers, as: :json
-
-        expect(response.code).to eq '200'
-        response_json = JSON.parse(response.body)
-        expect(response_json).to eq(
-          [
-            {
-              'names' => organization.names,
-              'tag_names' => organization.tag_names,
-              'website_urls' => organization.website_urls,
-              'cities' => organization.cities,
-              'countries' => organization.countries,
-              'id' => organization.id,
-              'in_business' => organization.in_business
-            }
-          ]
-        )
-      end
-
-      it 'returns no more results than the DEFAULT_SEARCH_SIZE' do
-        default_size = Api::V1::SearchController::DEFAULT_SEARCH_SIZE
-
-        Organization.recreate_index!
-
-        (default_size + 1).times do
-          organization = Fabricate :organization
-          Fabricate :organization_name, organization: organization
-          organization.es_index
-        end
-
-        Organization.refresh_index!
-
-        get '/v1/search', params: { term: 'Gallery' }, headers: headers, as: :json
-
-        response_json = JSON.parse(response.body)
-        expect(response_json.count).to eq default_size
-      end
-
-      it 'respects the size sent in from the client' do
-        default_size = Api::V1::SearchController::DEFAULT_SEARCH_SIZE
-        Organization.recreate_index!
-
-        default_size.times do
-          organization = Fabricate :organization
-          Fabricate :organization_name, organization: organization
-          organization.es_index
-        end
-
-        Organization.refresh_index!
-
-        size_override = default_size - 1
-        get '/v1/search', params: { term: 'Gallery', size: size_override }, headers: headers, as: :json
```

Comment:
I think there's a regression here around the number of organizations returned - is there a GraphQL convention for this concept that we can use?

Reply:
Are you asking for conventions around pagination? If so, this is what FB and GitHub use https://facebook.github.io/relay/graphql/connections.htm and we started using in MP later on. It’s all about using cursors, basically, and adds a convention for how to add metadata around a collection.

Reply:
Yeah, that's good, but what I did for the API that this is replacing was support a param that decided how many records to return. The autocomplete won't need more than, say, 20 records. Do you have a way you think that should work? I guess you could use the cursor stuff, but is there an easier way?

Reply:
That's fine, [this query](http://metaphysics-staging.artsy.net/?query=%7B%0A%20%20artist(id%3A%22banksy%22)%20%7B%0A%20%20%20%20artworks_connection(first%3A5)%20%7B%0A%20%20%20%20%20%20edges%7B%0A%20%20%20%20%20%20%20%20node%20%7B%0A%20%20%20%20%20%20%20%20%20%20title%0A%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%7D%0A%20%20%20%20%7D%0A%20%20%7D%0A%7D&operationName=null) would return the first 5 elements on the connection 

```graphql
{
  artist(id:"banksy") {
    artworks_connection(first:5) {
      edges{
        node {
          title
        }
      }
    }
  }
}
```

Sidenote: We don't do this, but the FB reference GraphQL implementation [has both](http://graphql.org/swapi-graphql/?query=%7B%0A%20%20allFilms%20%7B%0A%20%20%20%20edges%20%7B%0A%20%20%20%20%20%20node%20%7B%0A%20%20%20%20%20%20%20%20title%0A%20%20%20%20%20%20%7D%0A%20%20%20%20%7D%0A%20%20%20%20films%20%7B%0A%20%20%20%20%20%20title%0A%20%20%20%20%7D%0A%20%20%7D%0A%7D&operationName=null) `{ edges: { node } }` for relay/apollo and a named version for convenience.

Reply:
Oh perfect, thanks! So then @dblock, what do you think about adding a spec for this?

Reply:
I added support for `first` with a spec.